### ISSUE TITLE
Fix false positives for SVG type selectors in selector-type-case

### ISSUE THREAD

**Issue by hverlin** [stylelint/stylelint#5712]
### What steps are needed to reproduce the bug?

There is a false positive for `foreignObject`. 

The following code will indicate that there is `selector-type-case` issue
```
foreignObject {
  color: white;
}
```

Link to a [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBmEBOMCWBzAdgeQCMArGMAFwAJgAdHCiyAGzXgoHcALDMmAbloC+tEABoQAZwCeOMgEMAHgnDjxo8BBxJsSmnQrUQqAK6MY4g61316B8TFPk0AWjKSADjCdhZdixQDaBsxsMKgGALq09EI4AiACQA)

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-type-case": ["lower"]
  }
}
```

### How did you run Stylelint?

CLI, or see the demo link

### Which version of Stylelint are you using?

`14.0.0`, `13.13.1`

### What did you expect to happen?

No problems to be reported

### What actually happened?

Got the following error:
```
Expected "foreignObject" to be "foreignobject" (selector-type-case)
```

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

There should be no issue as [`foreignObject` tag is part of SVG definition](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject).

A potential workaround for now is to use the following config
```json
{
  "rules": {
    "selector-type-case": ["lower", {
      "ignoreTypes": ["foreignObject"]
    }]
  }
}
```

**Comment by ybiquitous** [comment]
@hverlin Thank you for using a template! It seems useful to ignore `foreignObject` (something else?) by default.

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

Perhaps, a bugfix may add an additional check to `tag.value` before the following code:

https://github.com/stylelint/stylelint/blob/bf34554b8a1fc0f7a06d0b0f0279b7c9273a0634/lib/rules/selector-type-case/index.js#L62-L64

**Comment by ybiquitous** [comment]
When seeing <https://developer.mozilla.org/en-US/docs/Web/SVG/Element>, there are many elements with mixed-case like `animateMotion`. 👀

**Comment by jeddy3** [comment]
Let's add these to our reference file and ignore them by default.

**Comment by hverlin** [comment]
Thanks! Will create a PR with the suggested fix

**Comment by hverlin** [comment]
Created #5717, let me know if you have feedback.
